crates/protocol/src/message/close.rs

```diff
@@ -0,0 +1,81 @@
+use ::serde::{Deserialize, Serialize};
+use chrono::Utc;
+use type_safe_id::{DynamicType, TypeSafeId};
+
+use super::{Message, MessageError, MessageKind, MessageMetadata};
+
+pub struct Close;
+
+impl Close {
+    pub fn create(
+        from: String,
+        to: String,
+        exchange_id: TypeSafeId<DynamicType>,
+        reason: Option<String>,
+    ) -> Result<Message<CloseData>, MessageError> {
+        let metadata = MessageMetadata {
+            from,
+            to,
+            kind: MessageKind::Close,
+            id: MessageKind::Close.typesafe_id()?,
+            exchange_id: exchange_id,
+            created_at: Utc::now(),
+        };
+
+        let data = CloseData { reason };
+
+        Ok(Message {
+            metadata,
+            data,
+            signature: None,
+        })
+    }
+}
+
+/// A struct representing the data contained within the [`Message`] for a [`Close`].
+///
+/// See [Quote](https://github.com/TBD54566975/tbdex/tree/main/specs/protocol#close) for more
+/// information.
+#[derive(Debug, Deserialize, PartialEq, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct CloseData {
+    /// an explanation of why the exchange is being closed/completed
+    reason: Option<String>,
+}
```

Comment:
Just realized, we _might_ want to not serialize empty fields. I'm not sure if this behavior is defined in the current tbDEX spec or not. On first glance, I don't think there's been a decision made.

Regardless, there's two ways I know of where we _could_ skip serializing empty optional fields:
* Annotate the optional field with `#[serde(skip_serializing_if = "Option::is_none")]`
* Use the `serde_with` crate, which comes with a nice [#[skip_serializing_none]](https://docs.rs/serde_with/latest/serde_with/attr.skip_serializing_none.html) macro

For reference, we're using the `serde_with` crate over in `web5-rs`, as it makes the code a little bit less messy, and maintains the readability a bit nicer.

Don't think we should take action on this right now, but I'll go ahead and make a task in the backlog for this.

Reply:
Good point! IMO we should not serialize `None` fields. I pushed a commit adding `serde_with` since it was a pretty easy additional. If we change our minds I'll strip it out.